applications/protocols_serialization/client/src/ot_shell.c

```diff
@@ -19,16 +23,244 @@ static int ot_console_cb(void *context, const char *format, va_list arg)
 	return 0;
 }
 
-static int ot_cmd(const struct shell *sh, size_t argc, char *argv[])
+static void ot_cli_lazy_init(const struct shell *sh)
+{
+	static bool is_initialized;
+
+	if (is_initialized) {
+		return;
+	}
+
+	otCliInit(NULL, ot_cli_output_cb, (void *)sh);
+	is_initialized = true;
+}
+
+typedef otError (*ot_cli_command_handler_t)(const struct shell *, size_t argc, char *argv[]);
+
+/*
+ * Adapter function for processing OpenThread shell commands, which is convenient to call from
+ * a Zephyr shell command handler. It is meant to duplicate the behavior of the internal OT CLI
+ * component with regards to printing the command's result.
+ */
+static int ot_cli_command_invoke(ot_cli_command_handler_t handler, const struct shell *sh,
+				 size_t argc, char *argv[])
+{
+	otError error;
+
+	ot_cli_lazy_init(sh);
+	error = handler(sh, argc, argv);
+
+	if (error == OT_ERROR_NONE) {
+		shell_print(sh, "Done");
+	} else {
+		shell_print(sh, "Error %d", error);
+	}
+
+	return 0;
```

Comment:
Shouldn't it return any error code in case when handler returns error?

Reply:
Does this change any behavior? The shell core does not display any error code on error etc. from my experience so I'm not even sure what the error code does in the shell handler.

Reply:
Yes, it does. https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/shell/shell.h#L162-L175 api description.
[shell_get_return_value](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/shell/shell.h#L1292-L1299)

Reply:
ok, fixed.